src/push-state-tree.js

```diff
@@ -106,49 +156,63 @@ var elementPrototype = typeof HTMLElement !== 'undefined' ? HTMLElement : Elemen
 
 function PushStateTree(options) {
   options = options || {};
-  options[USE_PUSH_STATE] = options[USE_PUSH_STATE] !== false;
 
   // Force the instance to always return a HTMLElement
   if (!(this instanceof elementPrototype)) {
     return PushStateTree.apply(PushStateTree.createElement('pushstatetree-route'), arguments);
   }
 
-  this.VERSION = VERSION;
+  this.eventStack = {
+    leave: [],
+    change: [],
+    enter: [],
+    match: []
+  };
 
-  // Setup options
-  for (var prop in options) {
-    if (options.hasOwnProperty(prop)) {
-      this[prop] = options[prop];
+  //TODO: emcapsulate this
+  for (let property in PushStateTree.prototype) {
+    if (typeof PushStateTree.prototype[property] === 'function') {
+      // function wrapper, without bind the context
+      this[property] = PushStateTree.prototype[property].bind(this);
+      continue;
+    }
+    // Copy properties from prototype to the instance
+    if (typeof this[property] == 'undefined') {
+      this[property] = PushStateTree.prototype[property];
     }
   }
```

Comment:
Yes, it's because the PST uses DOM element as instance, so it applies the same technique from Polymer project, by bounding methods/properties from a prototype into a instance.

The initial idea of doing this was to allow usage of DOM elements to not need to implement event listeners, and make all the DOM resources available (like querystring, DOM exposes, and others).

However by giving you this explanation I understand I did a mistake by changing the old wrapper by bind. The old wrapper was accessing the prototype, witch means if the prototype were changed, it would affect all instances allowing plugins to manipulate it. The way it's right now, the prototype changes won't affect the instances.

This is a good place to add test case and fix it. Thanks!